### ISSUE TITLE
Bug / Feature Request: Brightness and color control not working properly

### ISSUE THREAD

**Issue by Cheerpipe** [t0bst4r/matterbridge-home-assistant#49]
Hi.

I have several wled and yeelight lights that are exposed only as on/off devices.

Yeelight light attributes

```yaml
min_color_temp_kelvin: 1700
max_color_temp_kelvin: 6535
min_mireds: 153
max_mireds: 588
effect_list:
  - Strobe color
  - Police
  - Christmas
  - RGB
  - Random Loop
  - Fast Random Loop
  - LSD
  - Slowdown
  - Night Mode
  - Date Night
  - Movie
  - Sunrise
  - Sunset
  - Romance
  - Happy Birthday
  - Disco
  - Strobe epilepsy!
  - Alarm
  - Police2
  - WhatsApp
  - Facebook
  - Twitter
  - Home
  - Candle Flicker
  - Tea Time
  - Slow Temp
  - Stop
supported_color_modes:
  - color_temp
  - hs
  - rgb
effect: null
color_mode: null
brightness: null
color_temp_kelvin: null
color_temp: null
hs_color: null
rgb_color: null
xy_color: null
flowing: false
music_mode: false
icon: mdi:ceiling-light
friendly_name: Comedor techo
supported_features: 44
```

Wled lights attributes 

```yaml 
effect_list:
  - Akemi
  - Android
  - Aurora
  - Black Hole
  - Blends
  - Blink
  - Blink Rainbow
  - Blobs
  - Blurz ☾
  - Bouncing Balls
  - Bpm
  - Breathe
  - Candle
  - Candle Multi
  - Chase
  - Chase 2
  - Chase 3
  - Chase Flash
  - Chase Flash Rnd
  - Chase Rainbow
  - Chase Random
  - Chunchun
  - Colored Bursts
  - Colorful
  - Colorloop
  - Colortwinkles
  - Colorwaves
  - Crazy Bees
  - DJ Light
  - DNA
  - DNA Spiral
  - Dancing Shadows
  - Dissolve
  - Dissolve Rnd
  - Distortion Waves
  - Drift
  - Drift Rose
  - Drip
  - Dynamic
  - Dynamic Smooth
  - Fade
  - Fairy
  - Fairytwinkle
  - Fill Noise
  - Fire 2012
  - Fire Flicker
  - Firenoise
  - Fireworks
  - Fireworks 1D
  - Fireworks Starburst
  - Fireworks audio ☾
  - Flow
  - Flow Stripe
  - Freqmap
  - Freqmatrix
  - Freqpixels
  - Freqwave
  - Frizzles
  - Funky Plank
  - Fw Starburst audio ☾
  - GEQ ☾
  - Game Of Life
  - Ghost Rider
  - Glitter
  - Gradient
  - Gravcenter
  - Gravcentric
  - Gravfreq ☾
  - Gravimeter ☾
  - Halloween Eyes
  - Heartbeat
  - Hiphotic
  - ICU
  - Juggle
  - Juggles
  - Julia
  - Lake
  - Lighthouse
  - Lightning
  - Lissajous ☾
  - Loading
  - Matripix ☾
  - Matrix
  - Metaballs
  - Meteor
  - Meteor Smooth
  - Midnoise
  - Multi Comet
  - Noise 1
  - Noise 2
  - Noise 3
  - Noise 4
  - Noise Pal
  - Noise2D
  - Noisefire
  - Noisemeter
  - Noisemove
  - Octopus
  - Oscillate
  - Pacifica
  - Palette
  - Party jerk
  - Percent
  - Perlin Move
  - Phased
  - Phased Noise
  - Pixels
  - Pixelwave
  - Plasma
  - Plasma Ball
  - Plasmoid
  - Polar Lights
  - Popcorn
  - Popcorn audio ☾
  - Pride 2015
  - Puddlepeak
  - Puddles
  - Pulser
  - RSVD
  - RSVD
  - RSVD
  - RSVD
  - RSVD
  - RSVD
  - RSVD
  - RSVD
  - RSVD
  - RSVD
  - RSVD
  - Railway
  - Rain
  - Rainbow
  - Rainbow Runner
  - Random Colors
  - Ripple
  - Ripple Peak
  - Ripple Rainbow
  - Rocktaves
  - Rolling Balls
  - Running
  - Running Dual
  - Saw
  - Scan
  - Scan Dual
  - Scanner
  - Scanner Dual
  - Scrolling Text
  - Sindots
  - Sine
  - Sinelon
  - Sinelon Dual
  - Sinelon Rainbow
  - Soap
  - Solid
  - Solid Glitter
  - Solid Pattern
  - Solid Pattern Tri
  - Spaceships
  - Sparkle
  - Sparkle Dark
  - Sparkle+
  - Spots
  - Spots Fade
  - Squared Swirl
  - Stream 2 ☾
  - Stream ☾
  - Strobe
  - Strobe Mega
  - Strobe Rainbow
  - Sun Radiation
  - Sunrise
  - Sweep
  - Sweep Random
  - Swirl
  - TV Simulator
  - Tartan
  - Tetrix
  - Theater
  - Theater Rainbow
  - Traffic Light
  - Tri Fade
  - Tri Wipe
  - Twinkle
  - Twinklecat
  - Twinklefox
  - Twinkleup
  - Two Dots
  - Washing Machine
  - Waterfall
  - Waverly ☾
  - Wavesins
  - Waving Cell
  - Wipe
  - Wipe Random
  - ⚙️ ARTI-FX ☾
supported_color_modes:
  - rgb
effect: null
color_mode: null
brightness: null
hs_color: null
rgb_color: null
xy_color: null
icon: mdi:television-ambient-light
friendly_name: Televisor
supported_features: 36
```

**Comment by t0bst4r** [comment]
Hey @Cheerpipe ,

I had a quick look at the [developer documentation](https://developers.home-assistant.io/docs/core/entity/light/#color-modes) and saw that my implementation is actually wrong. Brightness is only set in the `supported_color_modes` if no other color feature is supported. 🤦🏼‍♂️
If any other feature is supported, brightness is implicit.

So actually this ticket is about properly supporting RGB and ColorTemp. And about fixing brightness for hs.

Thank you for reaching out.

**Comment by Cheerpipe** [comment]
Hi 
Glad to be of help.

**Comment by Cheerpipe** [comment]
light.turn_on RGB call example

```
service: light.turn_on (with this some bulbs change to rgb mode and others to hs but works for both)
data:
  rgb_color:
    - 101
    - 37
    - 37
target:
  entity_id: light.dormitorio_pequeno_techo
```

light.turn_on color_temp call example (It sets my bulbs to color_temp mode

```
service: light.turn_on
data:
  color_temp: 409
target:
  entity_id: light.dormitorio_pequeno_techo
```

```
light.turn_on hs_color call example (It sets my bulbs to color_temp mode
service: light.turn_on
data:
  hs_color: [0.12,0.98]
target:
  entity_id: light.dormitorio_pequeno_techo
```

**Comment by t0bst4r** [comment]
I learned a lot in the last days how matter and home assistant handle hue and saturation differently. I worked on a fix to properly detect color support and to set the values correct. 
HS- and RGB color are supported, then. I did not understand how xy color work with Home Assistant since it does not match the known XYZ color spaces on any way. 
color temperature will also be supported. Since every color supported light gets color temperature controls via matter, i created a workaround to convert color temp to HS if color temp is not supported by the HA light. 

I’ll release all this tomorrow

**Comment by t0bst4r** [comment]
@Cheerpipe 
What I have found out so far is that `hs_color` is set using hue (0-360) and saturation (0-100).
I am wondering about your last example setting hs to `[ 0.12, 0,98 ]`. I assume that this is percentage.

Does HA automatically detect if percentage or absolute values are used?

Could you do some tests here?
E.g. by setting `[ 0.5, 0.5 ]` and comparing it to `[ 180, 50 ]` - or other values?

**Comment by t0bst4r** [comment]
Just released version 0.4.0 including those changes. Please check and provide feedback 😁
